src/ProfileEndpoints.jl

```diff
@@ -242,6 +245,30 @@ end
 
 end  # if isdefined
 
+###
+### Type Inference
+###
+
+function typeinf_start_endpoint(req::HTTP.Request)
+    Core.Compiler.__set_measure_typeinf(true)
+    return HTTP.Response(200, "Type inference profiling started.")
+end
+
+function typeinf_stop_endpoint(req::HTTP.Request)
+    Core.Compiler.__set_measure_typeinf(true)
+    timings = if isdefined(Core.Compiler.Timings, :clear_and_fetch_timings)
+        # TODO: make a root node out of this
+        Core.Compiler.Timings.clear_and_fetch_timings()
+    else
+        InferenceTimingNode(Core.Compiler.Timings._timings[1])
+    end
```

Comment:
Same here actually -- i think you can directly call `SnoopCompileCore. stop_deep_timing(); SnoopCompileCore.finish_snoopi_deep()`, instead of having to rebuild it yourself, and this way the different API logic is contained to just that one package and you don't need to check `isdefined` stuff.